src/app/pages/customer/notifications/notificaciones/notificaciones.component.ts

```diff
@@ -51,39 +51,53 @@ export class NotificacionesComponent implements OnInit {
     }
   }
 
-
   saveNotifications() {
     localStorage.setItem(`notifications_${this.usuarioActual}`, JSON.stringify(this.notifications));
   }
 
-
   confirmNotification(index: number) {
-    // Eliminar la notificación del arreglo y actualizar localStorage
     this.notifications.splice(index, 1);
     this.saveNotifications();
+    this.clearInterval(index); // Limpiar intervalo relacionado
   }
 
-
-  scheduleNotifications() {
-    this.notifications.forEach((notification) => {
+  initializeNotificationIntervals() {
+    this.notifications.forEach((notification, index) => {
       if (notification.interval) {
-        setTimeout(() => {
-          // Agregar una nueva notificación al final del intervalo
-          this.notifications.push({
-            message: notification.message,
-            time: new Date(),
-            interval: notification.interval,
-          });
-          this.saveNotifications();
-
-
-          // Reprogramar el mismo recordatorio
-          this.scheduleNotifications();
-        }, notification.interval);
+        this.setNotificationInterval(notification, index);
       }
     });
   }
 
+  setNotificationInterval(notification: Notification, index: number) {
+    if (this.notificationIntervals.has(index)) {
```

Comment:
La verificación con this.notificationIntervals.has(index) es un buen enfoque para evitar intervalos duplicados. Esto muestra un entendimiento sólido del manejo de recursos en JavaScript